### PR TITLE
Switch to using `setTimeout` for all deferred processing.

### PR DESCRIPTION
**Pull request by thetorpedodog**
As noted on issue #392, using `requestAnimationFrame` means that no deferred tasks can take place while the user has the browser window in the background. This can lead to a bad experience for long-lasting tasks, particularly setup tasks (when a user might start JupyterLab, switch somewhere else when it loads, and then come back).

This change instead uses `setTimeout` (with a timeout of zero, either implicitly or explicitly), which immediately puts the task on the event loop to be run (subject to less severe browser throttling). This allows these tasks to occur in the background (and should hopefully not be too taxing on user resources).

https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#timeouts_in_inactive_tabs

> Firefox Desktop and Chrome both have a minimum timeout of 1 second > for inactive tabs.

### CODE REVIEW

**Comment by thetorpedodog** [comment]
@meeseeksdev tag bug

File: packages/polling/src/poll.ts [hunk 1]
```diff
@@ -347,7 +326,13 @@ export class Poll<T = any, U = any, V extends string = 'standby'>
...
   private _tick = new PromiseDelegate<this>();
   private _ticked = new Signal<this, IPoll.State<T, U, V>>(this);
-  private _timeout: any = -1;
+  /*
+   * If set, the timeout handle for this poller. We use this complex formulation
+   * for the type since the NodeJS environment has a different return type
+   * (`NodeJS.Timeout`) from the browser environment (`number`) and we want this
+   * to compile in both places.
+   */
+  private _timeout?: ReturnType<typeof setTimeout>;
```
Comment (fcollonval) [inline packages/polling/src/poll.ts:335]: nice

File: packages/polling/src/poll.ts [hunk 2]
```diff
@@ -241,17 +224,18 @@ export class Poll<T = any, U = any, V extends string = 'standby'>
     this._tick = scheduled;
 
     // Clear the schedule if possible.
-    if (last.interval === Poll.IMMEDIATE) {
-      cancel(this._timeout);
-    } else {
-      clearTimeout(this._timeout);
-    }
+    clearTimeout(this._timeout);
```
Comment (fcollonval) [inline packages/polling/src/poll.ts:227]: Should we test for non-undefined timeout id?

```suggestion
    if (typeof this._timeout === 'number') {
      clearTimeout(this._timeout);
    }
```

**Review by fcollonval — APPROVED** [review]
Thanks @thetorpedodog

**Comment by fcollonval** [comment]
We probably want to backport this to Lumino 1.x to include it in a JupyterLab patch release

**Comment by thetorpedodog** [comment]
in this case I will pull it out of the drafts so it can be merged!

File: packages/polling/src/poll.ts (hunk 1 above)
Comment (thetorpedodog) [reply packages/polling/src/poll.ts:335]: for the record I [learned this from Stack Overflow](https://stackoverflow.com/a/56239226/39808)

**Comment by thetorpedodog** [comment]
> We probably want to backport this to Lumino 1.x to include it in a JupyterLab patch release

Just did a test and the patch applies cleanly, so this can be automated post-submission: https://github.com/TileDB-Inc/lumino-dev/commit/1c2e89eb31bf9edd09c2db05d9cb192cb3bebb49

**Comment by afshin** [comment]
Thank you, @thetorpedodog! Great work tracking down this issue from several layers downstream.

I changed your comment on the `_timeout` variable to make a terse in-line comment instead of a docstring style just to keep it more idiomatic with the rest of the library, I hope that's okay!

File: packages/polling/src/poll.ts (hunk 1 above)
Comment (afshin) [inline packages/polling/src/poll.ts:335]:
```suggestion
  private _timeout?: ReturnType<typeof setTimeout>; // Support node and browser.
```

File: packages/polling/src/poll.ts [hunk 3]
```diff
@@ -416,7 +401,8 @@ export namespace Poll {
     standby?: Standby | (() => boolean | Standby);
   }
   /**
-   * An interval value that indicates the poll should tick immediately.
+   * An interval value that indicates the poll should tick immediately,
+   * i.e. 0 msec.
```
Comment (afshin) [inline packages/polling/src/poll.ts:405]:
```suggestion
   * An interval value (0ms) that indicates the poll should tick immediately.
```

**Review by afshin — APPROVED** [review]

**Comment by blink1073** [comment]
@meeseeksdev please backport to 1.x

**Pull request merged**